providers/test/Test.fst

```diff
@@ -359,126 +374,347 @@ let test_dh () : St unit =
   // TODO supposed to use tactics now?
   ()
 
-#reset-options "--z3rlimit 40"
+inline_for_extraction
+noeq
+type features = {
+  features_avx: bool;
+  features_avx2: bool;
+  features_bmi2: bool;
+  features_adx: bool;
+  features_aesni: bool;
+  features_shaext: bool;
+}
+
+inline_for_extraction
+let f_concat (f1 f2: features) : Tot features =
+  [@inline_let]
+  let avx = f1.features_avx || f2.features_avx in
+  [@inline_let]
+  let avx2 = f1.features_avx2 || f2.features_avx2 in
+  [@inline_let]
+  let bmi2 = f1.features_bmi2 || f2.features_bmi2 in
+  [@inline_let]
+  let adx = f1.features_adx || f2.features_adx in
+  [@inline_let]
+  let aesni = f1.features_aesni || f2.features_aesni in
+  [@inline_let]
+  let shaext = f1.features_shaext || f2.features_shaext in
+  {
+    features_avx = avx;
+    features_avx2 = avx2;
+    features_bmi2 = bmi2;
+    features_adx = adx;
+    features_aesni = aesni;
+    features_shaext = shaext;
+  }
+
+inline_for_extraction
+let f_none : features = {
+  features_avx = false;
+  features_avx2 = false;
+  features_bmi2 = false;
+  features_adx = false;
+  features_aesni = false;
+  features_shaext = false;
+}
+
+inline_for_extraction
+let f_avx : features = // [@inline_let] ({ f_none with features_avx = true; })
+{
+  features_avx = true;
+  features_avx2 = false;
+  features_bmi2 = false;
+  features_adx = false;
+  features_aesni = false;
+  features_shaext = false;
+}
+
+inline_for_extraction
+let f_avx2 : features = // [@inline_let] ({ f_none with features_avx2 = true; })
+{
+  features_avx = false;
+  features_avx2 = true;
+  features_bmi2 = false;
+  features_adx = false;
+  features_aesni = false;
+  features_shaext = false;
+}
+
+inline_for_extraction
+let f_bmi2 : features = // [@inline_let] ({ f_none with features_bmi2 = true; })
+{
+  features_avx = false;
+  features_avx2 = false;
+  features_bmi2 = true;
+  features_adx = false;
+  features_aesni = false;
+  features_shaext = false;
+}
+
+inline_for_extraction
+let f_adx : features = // [@inline_let] ({ f_none with features_adx = true; })
+{
+  features_avx = false;
+  features_avx2 = false;
+  features_bmi2 = false;
+  features_adx = true;
+  features_aesni = false;
+  features_shaext = false;
+}
+
+inline_for_extraction
+let f_aesni : features = // [@inline_let] ({ f_none with features_aesni = true; })
+{
+  features_avx = false;
+  features_avx2 = false;
+  features_bmi2 = false;
+  features_adx = false;
+  features_aesni = true;
+  features_shaext = false;
+}
+
+inline_for_extraction
+let f_shaext : features = // [@inline_let] ({ f_none with features_shaext = true; })
+{
+  features_avx = false;
+  features_avx2 = false;
+  features_bmi2 = false;
+  features_adx = false;
+  features_aesni = false;
+  features_shaext = true;
+}
+
+inline_for_extraction
+type impl = | Hacl | Vale | OpenSSL | BCrypt
```

Comment:
I think we should get rid of OpenSSL / BCrypt -- they're here only for legacy purposes and we're getting rid of those legacy functions very soon

Reply:
Thanks for confirming.
I initially thought that we should test OpenSSL and BCrypt for the purpose of testing the "bindings" between EverCrypt and existing libraries.